dds_web/commands.py

```diff
@@ -818,75 +826,250 @@ def quarterly_usage():
+    )
+
+    end = current_time()
+    flask.current_app.logger.debug(f"Month now: {end.month}")
+
+    start = end - relativedelta(months=months)
+    flask.current_app.logger.debug(f"Month {months} months ago: {start.month}")
+
+    flask.current_app.logger.debug(f"Start: {start}")
+    flask.current_app.logger.debug(f"End: {end}")
+
+    # CSV files to send
+    csv_file_names = []
+
+    # Iterate through units
+    for unit in models.Unit.query:
+        # CSV file
+        csv_file_name = pathlib.Path(
+            f"{unit.public_id}_Usage_Months-{start.month}-to-{end.month}.csv"
+        )
+        flask.current_app.logger.debug(f"CSV file name: {csv_file_name}")
+        csv_file_names.append(csv_file_name)
+
+        # Total usage for unit
+        total_usage = 0
+
+        # Open new csv file
+        try:
+            with csv_file_name.open(mode="w+", newline="") as file:
+                csv_writer = csv.writer(file)
+                csv_writer.writerow(
+                    [
+                        "Project ID",
+                        "Project Title",
+                        "Project Created",
+                        "Time Collected",
+                        "Byte Hours",
+                    ]
+                )
+
+                # Get usage rows connected to unit, that have been collected between X months ago and now
+                for usage_row, project_row in page_query(
+                    db.session.query(models.Usage, models.Project)
+                    .join(models.Project)
+                    .filter(
+                        models.Project.responsible_unit == unit,
+                        models.Usage.time_collected.between(start, end),
+                    )
+                ):
+                    # Increase total unit usage
+                    total_usage += usage_row.usage
+
+                    # Save usage row info to csv file
+                    csv_writer.writerow(
+                        [
+                            project_row.public_id,
+                            project_row.title,
+                            project_row.date_created,
+                            usage_row.time_collected,
+                            usage_row.usage,
+                        ]
+                    )
+
+                # Save total
+                csv_writer.writerow(["--", "--", "--", "--", total_usage])
+        except Exception as e:
+            flask.current_app.logger.error(f"Error writing to CSV file: {e}")
+            # delete already generated csv files
+            [csv_file.unlink() for csv_file in csv_file_names]
+            # Send email about error
+            email_message: flask_mail.Message = flask_mail.Message(
+                subject=error_subject,
+                recipients=[email_recipient],
+                body=error_body,
+            )
+            send_email_with_retry(msg=email_message)
+            raise
```

Comment:
This will raise the exception so none of the other unit files will be sent, so remove the raise.
```suggestion
```